GeneratorInterface/Core/src/EmbeddingHepMCFilter.cc

```diff
@@ -118,13 +119,16 @@ void EmbeddingHepMCFilter::decay_and_sump4Vis(HepMC::GenParticle *particle, reco
     // Asuming, that there are only the two tauons from the Z-boson.
     // This is the case for the simulated Z->tautau event constructed by EmbeddingLHEProducer.
     if (std::abs(particle->pdg_id()) == tauonPDGID_ && !decaymode_known) {
+      // use these bools to protect againt taus that aren't the last copy (which "decay" to a tau and a gamma)
+      bool isGamma = std::abs((*daughter)->pdg_id()) == 22;
+      bool isTau = std::abs((*daughter)->pdg_id()) == 15;
       if (std::abs((*daughter)->pdg_id()) == muonPDGID_) {
         DecayChannel_.fill(TauDecayMode::Muon);
         decaymode_known = true;
       } else if (std::abs((*daughter)->pdg_id()) == electronPDGID_) {
         DecayChannel_.fill(TauDecayMode::Electron);
         decaymode_known = true;
-      } else if (!neutrino) {
+      } else if (!neutrino && !isGamma && !isTau) {
         DecayChannel_.fill(TauDecayMode::Hadronic);
```

Comment:
@danielwinterbottom , if it is not tau, why it has tau hadronic decay mode?

Reply:
These bools are related to the tau daughters not the particle itself. L 121 already requires the particle to be a tau. This condition is used to ensure that we ignore taus that aren't the last copy in the chain which is the case when there is FSR and the tau "decays" to a tau and a gamma

Reply:
is this ok should I hit "resolve conversation" ?